thunder/executors/torchex.py

```diff
@@ -145,16 +145,10 @@ def _device_put_transform(a: TensorProxy, device: devices.Device) -> TensorProxy
 #
 
 
-class no_autocast(ContextDecorator):
-    def __enter__(self, *args, **kwargs):
-        self.was_cuda_enabled = torch.is_autocast_enabled()
-        self.was_cpu_enabled = torch.is_autocast_cpu_enabled()
-        torch.set_autocast_enabled(False)
-        torch.set_autocast_cpu_enabled(False)
-
-    def __exit__(self, exc_type, exc_value, exc_traceback):
-        torch.set_autocast_enabled(self.was_cuda_enabled)
-        torch.set_autocast_cpu_enabled(self.was_cpu_enabled)
+def no_autocast(fn):
+    fn = torch.autocast(device_type="cpu", enabled=False, cache_enabled=False)(fn)
+    fn = torch.autocast(device_type="cuda", enabled=False, cache_enabled=False)(fn)
+    return fn
```

Comment:
Could you add a test to prevent anyone from changing to a previous version?

Reply:
Let me try. Compiling a full LLaMA (which is what required this changes) is excessive for CI but I could try compiling a small sized LLaMA and try to assert that there was no `torch.compile()` recompilation